app/models/application_form.rb

```diff
@@ -72,7 +72,32 @@ def equality_and_diversity_answers_provided?
   attribute :recruitment_cycle_year, :integer, default: -> { RecruitmentCycle.current_year }
 
   before_create -> { self.support_reference ||= GenerateSupportRef.call }
-  after_save -> { application_choices.update_all(updated_at: Time.zone.now) }
+
+  PUBLISHED_FIELDS = %w[
+    first_name last_name support_reference phase submitted_at
+    becoming_a_teacher subject_knowledge interview_preferences
+    date_of_birth domicile right_to_work_or_study_details
+    english_main_language english_language_details other_language_details
+    disability_disclosure further_information safeguarding_issues_status
+    address_line1 address_line2 address_line3 address_line4
+    international_address country postcode equality_and_diversity
+    work_history_breaks first_nationality second_nationality third_nationality
+    fourth_nationality fifth_nationality phone_number
+  ].freeze
+
+  before_save do |form|
+    if (form.changed & PUBLISHED_FIELDS).any?
+      application_choices.touch_all
+    end
+  end
+
+  # We want to touch the application_choices when models that touch application_form are changed
+  # e.g. ApplicationReference, ApplicationQualification
+  # This ensures that the updated_at for the application changes when the API response for the application changes
+  after_touch do
```

Comment:
Justo to challenge my own assumption here: Do we always want to touch choices when forms are touched? This covers the mentioned references and qualifications. Do we maybe want a more precise way of doing this so that we can touch application forms without touching the choices? What do you think @duncanjbrown @steventux ?

Reply:
doesn't this line undo the work we do above? or is after_touch only called on explicit touches?

Reply:
This is only triggered on touch, either explicitly (`form.touch`) or via a belongs_to with a touch.

Reply:
I think the answer is that we do want to be more granular here. ApplicationFeedback has a `belongs_to :application form, touch: true`, but is not used in the API response for a choice. It will be clearer to add a concern that explicitly says "this model is published in the API"